### PR TITLE
Ported the Engine target to a shared library.

### PR DESCRIPTION
**Pull request by RubyNova**
Title.

### CODE REVIEW

File: cmake/Findglad.cmake [hunk 1]
```diff
@@ -53,7 +53,8 @@ if(glad_FOUND)
     COMMENT "Generating GLAD"
   )
 
-  add_library(glad "${glad_SOURCES}")
+  add_library(glad STATIC "${glad_SOURCES}")
+  set_target_properties(glad PROPERTIES POSITION_INDEPENDENT_CODE TRUE)
```
Comment (capnkenny) [inline cmake/Findglad.cmake:57]: We need to enable PIC now?

File: tests/NovelRT.Tests/CMakeLists.txt [hunk 2]
```diff
@@ -38,4 +38,30 @@ target_link_libraries(Engine_Tests
...
 )
 
+add_custom_command(
+  TARGET Engine_Tests POST_BUILD
+  COMMAND ${CMAKE_COMMAND} -E copy_directory
+    $<TARGET_FILE_DIR:Dotnet>
+    $<TARGET_FILE_DIR:Engine_Tests>/dotnet
+)
+
+#this is pure hacky hotfix goodness. We need to figure out a better way to do this in the future.
```
Comment (capnkenny) [inline tests/NovelRT.Tests/CMakeLists.txt:48]: Does this comment only apply to the Nethost workaround still? Or are you referring to all of the Post Build steps now?

File: cmake/Findglad.cmake (hunk 1 above)
Comment (RubyNova) [reply cmake/Findglad.cmake:57]: yes, because this change also affects Ubuntu.

File: tests/NovelRT.Tests/CMakeLists.txt (hunk 2 above)
Comment (RubyNova) [reply tests/NovelRT.Tests/CMakeLists.txt:48]: This is purely the win32 nethost workaround.

**Pull request merged**